habitat-lab/habitat/tasks/rearrange/actions/oracle_nav_action.py

```diff
@@ -3,30 +3,41 @@
 # LICENSE file in the root directory of this source tree.
 
 
+import magnum as mn
 import numpy as np
 from gym import spaces
 
 import habitat_sim
 from habitat.core.registry import registry
 from habitat.sims.habitat_simulator.actions import HabitatSimActions
-from habitat.tasks.rearrange.actions.actions import BaseVelAction
+from habitat.tasks.rearrange.actions.actions import (
+    BaseVelAction,
+    HumanoidJointAction,
+)
 from habitat.tasks.rearrange.utils import get_robot_spawns
 from habitat.tasks.utils import get_angle
 
 
 @registry.register_task_action
-class OracleNavAction(BaseVelAction):
+class OracleNavAction(BaseVelAction, HumanoidJointAction):
     """
     An action that will convert the index of an entity (in the sense of
-    `PddlEntity`) to navigate to and convert this to base control to move the
+    `PddlEntity`) to navigate to and convert this to base/humanoid joint control to move the
     robot to the closest navigable position to that entity. The entity index is
-    the index into the list of all available entities in the current scene.
+    the index into the list of all available entities in the current scene. The
+    config flag motion_type indicates whether the low level action will be a base_velocity or
+    a joint control.
     """
 
     def __init__(self, *args, task, **kwargs):
-        super().__init__(*args, **kwargs)
+        config = kwargs["config"]
+        self.motion_type = config.motion_control
+        if self.motion_type == "base_velocity":
+            BaseVelAction.__init__(self, *args, **kwargs)
+        else:
+            self.humanoid_controller = None
```

Comment:
See my other comment about making this more readable for the `human_joints` case.

Reply:
Thanks for the review! Provided better handling in latest commit.